compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs

```diff
@@ -565,6 +565,21 @@ data Expr
     , friRequiringInterface :: !(Qualified TypeConName)
     , friExpr :: !Expr
     }
+  -- | Obtain type representation of contract through interface
+  | EToTypeRep
+    { ttrInterface :: !(Qualified TypeConName)
+    , ttrExpr :: !Expr
+    }
+  -- | Obtain signatories of contract through interface
+  | EResolveVirtualSignatory
```

Comment:
I think these two primitives are the only ones that have resolve in the name. On the other hand, i think all other interface primitives are suffixed with `Interface`. How about calling these `ESignatoryInterface` and `EObserverInterface` and maybe also `EInterfaceTypeRep` or something like this?

Reply:
that sounds good. The only one I'm not sure about is `EInterfaceTypeRep`, since we actually want it to return the _template_ TypeRep. How about `EInterfaceInternalTypeRep`?

Reply:
`EInterfaceTemplateTypeRep`? Not sure what’s best

Reply:
ah I like that one, does exactly what it says. Thanks!